### ISSUE TITLE
Files in ignored directories are still being linted when individually listed

### ISSUE THREAD

**Issue by spadgos** [eslint/eslint#837]
This is basically the same issue as #813 (and its associated PR #818), but I've noticed a bug that made it through.

Situation:

```
/subdir
  my-file.js
.eslintignore
```

The `.eslintignore` file has the contents: `subdir`

``` bash
$ eslint subdir/my-file.js
```

**Expected**: a warning that my-file.js was ignored. **Actual**: my-file.js is linted.

This happens with all variations of ignoring a directory that I know of:

```
subdir
*/subdir
subdir/*
**/subdir/*
**/subdir/**
subdir/**
```

_(This is using [current master](https://github.com/eslint/eslint/tree/e3e1d4cad73bfdd84d8c7057dfd659724a791add) on Linux)_
